partitioner/src/main/java/io/confluent/connect/storage/partitioner/TimeBasedPartitioner.java

```diff
@@ -234,14 +236,107 @@ public void configure(Map<String, Object> config) {
       dateTime = ISODateTimeFormat.dateTime();
     }
 
+    private Object getField(Struct struct, String fieldName) {
+      Object field;
+      try {
+        field = struct.get(fieldName);
+      } catch (DataException e) {
+        throw new DataException(
+                String.format("The field named '%s' does not exist.", fieldName), e);
+      }
+      return field;
+    }
+
+    private Object getNestedFieldValue(Struct struct) {
+      final String[] fieldNames = fieldName.split("\\.");
+      Struct tmpStruct = struct;
+      Object tmpObject;
+      try {
+        // Iterate down to final struct
+        int i = 0;
+        while (i < fieldNames.length - 1) {
+          try {
+            tmpObject = getField(tmpStruct, fieldNames[i]);
+          } catch (DataException e) {
+            throw new DataException(
+                    String.format("Unable to find nested field '%s'", fieldNames[i]));
+          }
+          tmpStruct = (Struct) tmpObject;
+          i++;
+        }
+        // Extract from final struct
+        tmpObject = getField(tmpStruct, fieldNames[i]);
```

Comment:
I think this code can be slimmed down a bit.  I don't think you need to assign both a `tmpObject` and a `tmpStruct` in each iteration.  I think just the `tmpStruct` part should suffice to check for field name existence.  You can assign the resulting object after iteration is finished.

I'd also recommend a for loop here as it might look a bit cleaner.  That's more for aesthetics though.

Reply:
> I'd also recommend a for loop here as it might look a bit cleaner

I need `i` outside the scope of the loop, but `for ( ; i < fieldNames.length - 1; i++)` could be done, sure. 

You're probably correct about the `tmpObject`. Didn't give that much thought to optimization there.

Reply:
These methods require simplification and compaction. Given the tight deadlines we'd want to meet for the next release I'll be pushing my suggestions directly to this branch for this code part of the extraction of the nested fields and values.